lib/rules/no-inner-declarations.js

```diff
@@ -97,7 +111,15 @@ module.exports = {
 
         return {
 
-            FunctionDeclaration: check,
+            FunctionDeclaration(node) {
+                const isStrictFunction = sourceCode.getScope(node).isStrict;
```

Comment:
`getScope(node)` is the function's scope, but we should check the upper scope, because even if the function is strict code, when the declaration isn't in strict code the function is still not block-scoped.

```js
{
    function foo() {
        "use strict";
        console.log("foo called");
    }
}

foo(); // logs "foo called"

```

Also, it would be nice to check the scope last, for performance reasons. For example, the check can look like this:

```js
if (
    blockScopedFunctions === "allow" &&
    ecmaVersion >= 2015 &&
    sourceCode.getScope(node).upper.isStrict
) {
    return;
}
```

Reply:
Done!